client/proposer/service_test.go

```diff
@@ -51,15 +59,44 @@ func settingUpProposer(t *testing.T) (*Proposer, *internal.MockClient) {
 	if err != nil {
 		t.Fatalf("Failed to create proposer %v", err)
 	}
-	fakeProposer.config.CollationSizeLimit = int64(math.Pow(float64(2), float64(10)))
+	// params.DefaultConfig is a package level pointer var asigned to fakeProposer.config,
```

Comment:
hmm i don't feel very comfortable changing a singleton pointer this way :) it can have unexpected consequences as we see here. 

I've added a `GetDefaultConfig` factory to `param`, used it to get a fresh config for each `fakeProposer` and uncommented the code to modify specifically that `fakeProposer` config and anything else, all seems to be working fine now, PTAL.

Reply:
ah, k , yeah that is better probably shouldn't direct modify the `defaultconfig`.